packages/application-shell/src/components/authenticated/authenticated.tsx

```diff
@@ -1,23 +1,61 @@
+import type { ApplicationWindow } from '@commercetools-frontend/constants';
+import type { TAsyncLocaleDataProps } from '@commercetools-frontend/i18n';
+
 import React from 'react';
+import { Route, Switch } from 'react-router-dom';
 import hasCachedAuthenticationState from './has-cached-authentication-state';
 import AmILoggedIn from './am-i-logged-in';
+import AuthCallback from './auth-callback';
+
+declare let window: ApplicationWindow;
 
 type RenderFnArgs = { isAuthenticated: boolean };
 export type TProps = {
   render: (args: RenderFnArgs) => JSX.Element;
+  locale: string;
+  applicationMessages: TAsyncLocaleDataProps['applicationMessages'];
   children?: never;
 };
 
 const Authenticated = (props: TProps) => {
   // We attempt to see if the user was already authenticated by looking
   // at the "cached" flag in local storage.
   const cachedAuthenticationState = hasCachedAuthenticationState();
+
   if (cachedAuthenticationState) {
     return <>{props.render({ isAuthenticated: true })}</>;
   }
 
+  // When using the OIDC workflow, we always return false, to trigger
+  // the redirect to the login page.
+  if (window.app.__DEVELOPMENT__) {
+    return <>{props.render({ isAuthenticated: false })}</>;
+  }
+
+  // Fall back to check for cookie-based authentication.
   return <AmILoggedIn {...props} />;
 };
 Authenticated.displayName = 'Authenticated';
 
-export default Authenticated;
+const AuthenticationRoutes = (props: TProps) => (
+  <Switch>
+    <Route path={`/account/auth/callback`}>
```

Comment:
`auth` or `authorize`? Is it the same or is `auth` preferred?

Reply:
Good question.

Usually `/authorize` is well defined in the spec as the endpoint to start the authentication flow.
The `/auth/callback` is usually left to the user to decide.

I don't have a particular opinion about the name, it's also something we can manage internally.

Do you have suggestions? We can also change it to `/oidc/callback`

Reply:
I like `oidc/calback` to be honest. It clearly shows what auth flow the callback is for. Also if that deviates a bit from the norm.

Reply:
https://github.com/commercetools/merchant-center-application-kit/pull/1934/commits/496d5add7b13982ba5e2847555bf08e95e75db18